easydock/preparation_for_docking.py

```diff
@@ -35,43 +38,222 @@ def mol_from_smi_or_molblock(ligand_string):
 
 
 def mk_prepare_ligand(mol, verbose=False):
+    pdbqt_string_list = []
     preparator = MoleculePreparation(hydrate=False, flexible_amides=False, rigid_macrocycles=True, min_ring_size=7, max_ring_size=33)
     try:
-        mol_setups = preparator.prepare(mol)
-        for setup in mol_setups:
-            pdbqt_string, is_ok, error_msg = PDBQTWriterLegacy.write_string(setup)
-            if verbose:
-                print(f"{setup}")
+        cids = [x.GetId() for x in mol.GetConformers()]
+        for cid in cids:
+            mol_setups = preparator.prepare(mol, conformer_id=cid)
+
+            for setup in mol_setups:
+                pdbqt_string, is_ok, error_msg = PDBQTWriterLegacy.write_string(setup)
+                if not is_ok:
+                    print(f"{mol.GetProp('_Name')} has error in converting to pdbqt: {error_msg}")
+                pdbqt_string_list.append(pdbqt_string)
```

Comment:
What will be `pdbqt_string` if not `is_ok`? Is it reasonable to add this value to the list? Need to check

Reply:
In the case of the `PDBQTWriterError`, the `mk_prepare_ligand` return an empty string `''`. So, is it better to not append the empty string and let the program continue?

Reply:
Yes, it is not reasonable. Then it can be treated in this way below. If one conformer will fail, maybe some other will pass, so no need to interrupt preparation. In the worst case we will return an empty list.

```python
if not is_ok:
    print(f"{mol.GetProp('_Name')} has error in converting to pdbqt: {error_msg}")
else:
    pdbqt_string_list.append(pdbqt_string)
```